### ISSUE TITLE
Advice for improving generation.

### ISSUE THREAD

**Issue by Ruziy** [microsoft/mattergen#106]
Good day, developers. I have trained the chemical_system model on the density parameter, which I got from https://next-gen.materialsproject.org/materials, having previously combined them in the alex_mp_20 dataset by the material_id feature. But I have encountered a problem that when generating, when I try to increase or decrease the density, the crystal lattice of my material does not adapt to the specified parameter. Let me give you an example: I want to get the inconel material, which has 'chemical_system': 'Ni-Cr-Fe' and a density in the range from 7.8 to 8.2 depending on the temperature. But when I try to lower the density parameter to 7.6, the resulting density remains the same.
I get the results on the strength of the generated material using pymatgen. Model parameters ~100 epoch with metrics loss_train_step=0.315, pos_train_step=2.680, cell_train_step=0.00173, atomic_numbers_train_step=0.0458, loss_train_epoch=0.242, pos_train_epoch=1.930, cell_train_epoch=0.00745, atomic_numbers_train_epoch=0.0418] and loss_val=0.25.
I would like to get advice or recommendation on what to do in my situation, thank you.

**Comment by danielzuegner** [comment]
Hi @Ruziy,

without more details it's impossible for me to judge. My initial thoughts are

* Make sure that the new property is properly added to your dataset
* Make sure you're training on that new dataset
* Make sure that the training/ finetuning is adding your new property to the model
* Make sure you're correctly setting the property and diffusion guidance factor to 2.0

**Comment by Ruziy** [comment]
If any additional information is needed, I will be happy to provide it, thanks.

**Comment by Ruziy** [comment]
adding a command for fine tuning->
(.venv) (base) alexraz@DESKTOP-MG5O513:~/Desktop/mattergen$ export PROPERTY1=chemical_system
export PROPERTY2=density
export MODEL_NAME=chemical_system
(.venv) (base) alexraz@DESKTOP-MG5O513:~/Desktop/mattergen$ mattergen-finetune adapter.pretrained_name=$MODEL_NAME data_module=alex_mp_20 +lightning_module/diffusion_module/model/property_embeddings@adapter.adapter.property_embeddings_adapt.$PROPERTY1=$PROPERTY1 +lightning_module/diffusion_module/model/property_embeddings@adapter.adapter.property_embeddings_adapt.$PROPERTY2=$PROPERTY2 ~trainer.logger data_module.properties=["$PROPERTY1","$PROPERTY2"]

**Comment by danielzuegner** [comment]
Hi @Ruziy,

1. it looks like you're starting fine-tuning from the `chemical_system` pretrained model and pretrain it further. This is not something I've tried before so there may be subtle issues which cause problems. It's safest to start fine-tuning from the unconditional model.
2. It looks like you only have <5,000 labeled training samples. Since density can be computed trivially given an input structure, I recommend you fully label `alex_mp_20` and try again.

**Comment by Ruziy** [comment]
Hi @danielzuegner, 
1. OK, I'll try to train you to restart fine tuning on an unconditional model.
2. I showed the information from alex_mp_20 val.csv (to demonstrate the presence of the feature), the full dataset contains about ~56,000 instances with density.
To summarize, the problem is most likely with retraining on the chemical_system model. And should I try fine-tuning on the unconditional model? Are there any other suggestions? Thanks

**Comment by danielzuegner** [comment]
> Hi [@danielzuegner](https://github.com/danielzuegner),
> 
> 1. OK, I'll try to train you to restart fine tuning on an unconditional model.
> 2. I showed the information from alex_mp_20 val.csv (to demonstrate the presence of the feature), the full dataset contains about ~56,000 instances with density.
>    To summarize, the problem is most likely with retraining on the chemical_system model. And should I try fine-tuning on the unconditional model? Are there any other suggestions? Thanks

Since creating labels for density is very cheap, I still recommend training with 600k rather than 50k structures.

**Comment by Ruziy** [comment]
Yep, @danielzuegner I'm using the full alex_mp_20 dataset with 600k features, but only 50k of them have density, since it's only for Material Project data that I can get information about density. The remaining 540k have Nan.

**Comment by Mofahdi** [comment]
I am not sure if what I am about to share is helpful to you @Ruziy but you can get the densities of the other structures by writing something similar to the below:
from pymatgen.core.structure import Structure
import pandas as pd
df=pd.read_csv('val.csv')
for cif in df['cif'].values.tolist():
   density=Structure.from_str(cif).density

The above assumes that you have the cif files written in the csv file

**Comment by Ruziy** [comment]
Hi @Mofahdi, this could probably help, thanks, of course, there is an error in determining the density using this method. Do you think it might help @danielzuegner ?

**Comment by danielzuegner** [comment]
> Hi [@Mofahdi](https://github.com/Mofahdi), this could probably help, thanks, of course, there is an error in determining the density using this method. Do you think it might help [@danielzuegner](https://github.com/danielzuegner) ?

Hi @Ruziy, it looks like you're using this method to evaluate the density of your generated samples, so it makes sense to use the same method for creating the training labels for the model.

**Comment by Mofahdi** [comment]
@Ruziy If you have a cif file (from your generated structures), you can write: Structure.from_file('filename.cif')
if you have the cif file written inside the csv file (training dataset), you can read it like this: Structure.from_str(row['cif']) assuming you are iterating over the rows in the csv file and the column name is "cif"
I can refer you to the pymatgen format in the following link:
https://github.com/materialsproject/pymatgen/blob/master/src/pymatgen/core/structure.py
Hope this helps!

**Comment by Ruziy** [comment]
@Mofahdi thanks for helping, I'll try this approach.

**Comment by Ruziy** [comment]
Hi @danielzuegner I have a question about your model, dft_band_gap. As far as I could find data for training on this feature, there are about ~43,000 features, but you are presenting it as a demo working version. I'm wondering how you achieved this, it's just that while I was training a 50,000 feature model for density, I took the intermediate result after 100 epochs, and the problem remained. Now I'll try to retrain by filling in the gaps using pematgen.density. But the question of how you solved this problem is still interesting to me. Have you applied more eras of learning? As far as I can tell, it was about 100->500 epochs for normal operation. 

![Image](https://github.com/user-attachments/assets/d2beef94-cf98-4789-8868-045396422576)

**Comment by danielzuegner** [comment]
Hi @Ruziy,

we used the default settings from this repo to fine-tune the checkpoints. We fine-tuned the models for 200 epochs (as per the default settings). Have you tried fine-tuning from the unconditional base model?

**Comment by Ruziy** [comment]
I would also like to know the question of the mass fractions of the chemical material, if there is any solution in this regard. Conditionally, I want to generate inconel_718 with the chemical composition 'Ni-Cr-Fe-Nb-Ta-Mo-Ti'. But the percentages are obviously too different from the real ones. Where is the real thing: 

![Image](https://github.com/user-attachments/assets/f0151b05-d1ba-4a8e-a230-f486bf52e50a)
In turn, the generated ones have:

![Image](https://github.com/user-attachments/assets/dc0ce93e-d3b4-45f0-ab5b-ec85849a047a)

**Comment by Ruziy** [comment]
Complementing my problem, I would like to get the inconel element (the data about which is described above), but this data is not even in the training sample, which is strange because it is a fairly common metal. 

![Image](https://github.com/user-attachments/assets/e21fd40a-ce67-476e-a6a6-a5db0c6c2b05)
Perhaps @danielzuegner @Mofahdi you could help me by referring to more extensive datasets that contain data on this kind of material, on which I could further train the model (ideally, a direct link to the dataset source). Or offer some kind of solution of your own, is it possible to specify the mass fraction during generation? Or a priority for a specific chemical element. I would really appreciate it.

**Comment by Ruziy** [comment]
I raised this issue in a separate issue to reduce the number of recurring questions in the future.[124#issue](https://github.com/microsoft/mattergen/issues/124#issue-2963578387)
